hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -245,8 +246,10 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
           CreateBucketResponse.newBuilder().build());
       omClientResponse = new OMBucketCreateResponse(omResponse.build(),
           omBucketInfo, omVolumeArgs.copyObject());
-    } catch (IOException ex) {
-      exception = ex;
+    } catch (IOException | InvalidPathException ex) {
+      exception = ex instanceof IOException ? (IOException) ex :
+          new OMException(ex.getMessage(),
+              OMException.ResultCodes.INVALID_PATH);
       omClientResponse = new OMBucketCreateResponse(
           createErrorOMResponse(omResponse, exception));
```

Comment:
Can this exception translation logic be moved into `createErrorOMResponse`?  It would help avoid code duplication, and I think it's also better if the code using `exception` (logging, etc.) gets the original one.